to-do-app/src/App.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import './App.css';
+import TodoList from "./TodoList";
+
+class App extends Component {
+    constructor(){
+        super();
+        this.state = {
+            todos: [
+                {
+                    text: "Clean house",
+                    finished: true,
+                    id: Date.now()
+                },
+                {
+                    text: "Fix window",
+                    finished: false,
+                    id: Date.now()+1
+                }
+
+            ],
+            counter:0,
+            textArea:''
+        };
+    }
+    updateCounter(counter){
+        this.setState({counter:counter});
+    }
+    onTextClick(id){
+        const todos = this.state.todos.map((todo,i) => {
+            if(id==todo.id){ todo.finished = !todo.finished; }
+            return todo;
+        });
+        this.setState({todos});
+        this.countNotFinishedItems(todos);
+    }
+    onDeleteTodo(id){
+        const todos = this.state.todos.filter((todo) =>{ return todo.id !== id });
+        this.setState({todos});
+        this.countNotFinishedItems(todos);
+    }
+    countNotFinishedItems(todos){
+        const newTodos = todos || this.state.todos;
+        const counter = newTodos.filter((todo) =>{ return !todo.finished }).length;
+        this.updateCounter(counter);
+    }
+    componentDidMount() {
+        this.countNotFinishedItems();
+    }
+    onAddTodo(){
+        this.addTodo();
+    }
+    addTodo(text){
+        const todos = this.state.todos;
+        todos.push({
+            text: this.state.textArea,
+            finished: false,
+            id: Date.now()
+        });
+        this.countNotFinishedItems(todos);
+        this.setState({todos,textArea: ''});
+    }
+    onTextAreaChange(e){
+        this.setState({textArea: e.target.value});
+    }
+    render() {
+    return (
```

Comment:
remove count from the state
calculate in the render
const counter = newTodos.filter((todo) =>{ return !todo.finished }).length;